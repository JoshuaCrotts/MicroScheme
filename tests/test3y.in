//
// Tests global/local variables and mixed assignments.
//
// (NOTE: This does NOT work with Dr. Tate's interpreter - my modified one fixes the issue).
//

int x;
int y;

void printlnd(int d) {
	printd(d);
	prints("\n");
}

void main()
{
	int i = 753;
	int j = 444;

    x = 753;
    y = 444;
    printlnd(--x);
    printlnd(x = y--);
    
	printlnd(--i);
	printlnd(i = j--);
}
