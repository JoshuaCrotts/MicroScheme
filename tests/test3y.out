752
444
752
444
